qt/python/mantidqt/mantidqt/widgets/rawdataexplorer/view.py

```diff
@@ -0,0 +1,328 @@
+
+from mantid.simpleapi import *
+from mantid.kernel import logger
+from mantidqt.utils.qt import load_ui
+from mantidqt.widgets.instrumentview.api import *
+from mantidqt.widgets.sliceviewer.presenters.presenter import SliceViewer
+from mantidqt.plotting.functions import pcolormesh
+from workbench.config import get_window_config
+
+if TYPE_CHECKING:
+    from .presenter import PreviewPresenter, RawDataExplorerPresenter
+
+
+class PreviewView(QObject):
+
+    """
+    Type of the preview when the instrument viewer widget is used to show the
+    data.
+    """
+    IVIEW = "instrument_viewer"
+
+    """
+    Type of the preview when the slice viewer widget is used to show the data.
+    """
+    SVIEW = "slice_viewer"
+
+    """
+    Type of the preview when the 2d plot is used to show the data.
+    """
+    PLOT2D = "plot_2d"
+
+    """
+    Type of the preview when the 1d plot is used to show the data.
+    """
+    PLOT1D = "plot_1d"
+
+    """
+    Type of the preview when the spectrum plot is used to show the data.
+    """
+    PLOTSPECTRUM = "plot_spectrum"
+
+    """
+    Type the preview.
+    """
+    _type = None
+
+    """
+    Presenter.
+    """
+    _presenter = None
+
+    """
+    Reference to the visualisation widget.
+    """
+    _widget = None
+
+    """
+    Signal to request closing the preview
+    """
+    sig_request_close = Signal()
+
+    def __init__(self):
+        super().__init__()
+
+    def set_type(self, preview_type: str):
+        """
+        Set the preview type.
+        @param preview_type: the preview type
+        """
+        self._type = preview_type
+
+    def set_presenter(self, presenter: "PreviewPresenter"):
+        """
+        Set the presenter.
+        @param presenter: presenter
+        """
+        self._presenter = presenter
+
+    def show_workspace(self, workspace_name: str):
+        """
+        Show the workspace on the adapted widget.
+        @param workspace_name: the name of the workspace
+        """
+        try:
+            self.get_widget(workspace_name)
+        except ValueError as e:
+            message = "Could not open view:\n{0}".format(e)
+            logger.error(message)
+            message_box = QMessageBox()
+            message_box.setText(message)
+            message_box.exec()
+            self.on_close()
+            return
+
+        if self._type == self.IVIEW:
+            self._widget.show_view()
+            self._widget.container.closing.connect(self.on_close)
+            self.sig_request_close.connect(self._widget.container.emit_close)
+            self.sig_request_close.connect(lambda: self.on_close(None, False))
+
+        if self._type == self.SVIEW:
+            self._widget.view.show()
+            self._widget.view.close_signal.connect(self.on_close)
+            self.sig_request_close.connect(self._widget.view.emit_close)
+            self.sig_request_close.connect(lambda: self.on_close(None, False))
+
+        if self._type == self.PLOT2D:
```

Comment:
It seems that all plots have the exact same behaviour. Do you think they could be combined into one case?